src/main/java/raisetech/studentmanagement/controller/StudentController.java

```diff
@@ -5,18 +5,32 @@
 import org.springframework.web.bind.annotation.GetMapping;
 import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import raisetech.studentmanagement.controller.converter.StudentConverter;
 import raisetech.studentmanagement.data.Student;
 import raisetech.studentmanagement.data.StudentCourse;
+import raisetech.studentmanagement.domain.StudentDetail;
 import raisetech.studentmanagement.service.StudentService;
 
 @RestController
 public class StudentController {
 
-  private final StudentService service;
+  private StudentService service;
+  private StudentConverter converter;
 
   @Autowired
-  public StudentController(StudentService service) {
+  public StudentController(StudentService service, StudentConverter converter) {
     this.service = service;
+    this.converter = converter;
+  }
+
+  @GetMapping("/students/all")

```

Comment:
RESTで作るならallなどの名詞は不要かなと思います。
```suggestion
  @GetMapping("/students")

```

Reply:
エンドポイントを複数のメソッドに使うことは可能
クエストにクエリパラメータが含まれていればフィルタリングされた結果を、含まれていなければ全てのデータを返すように設定できます。
1つのエンドポイント内で条件分岐する方法が推奨されます。

Spring Bootがどちらのメソッドを使うべきかを判断できず、ビルドエラーが発生

Reply:
2.のstudents/coursesはコースが学生に紐づいているからこのパス設計なのですかね？
/courses
でいいような気がしました。

たとえばIDが123の学生にひもづくコースなら

/students/123/courses

のようなパス設計にしますね！

Reply:
> 2.のstudents/coursesはコースが学生に紐づいているからこのパス設計なのですかね？
>/courses
>でいいような気がしました。

Reply:
> RESTで作るならallなどの名詞は不要かなと思います。

修正いたしました！
70b925853447362af1753c4a9c1d069ff99e0a63

Reply:
> 3.条件分岐は検索に関する業務ロジックなので分岐をServiceに記述するほうが私的には嬉しいです。

条件分岐をServiceに記述しました！
70b925853447362af1753c4a9c1d069ff99e0a63